warehouse/query-core/src/test/java/datawave/query/jexl/visitors/FieldToFieldComparisonVisitorTest.java

```diff
@@ -0,0 +1,36 @@
+package datawave.query.jexl.visitors;
+
+import datawave.query.jexl.JexlASTHelper;
+import org.apache.commons.jexl2.parser.ASTEvaluationOnly;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class FieldToFieldComparisonVisitorTest {
+    @Test
+    public void testEq() throws ParseException {
+        ASTJexlScript query = FieldToFieldComparisonVisitor.forceEvaluationOnly(JexlASTHelper.parseJexlQuery("FOO == BAR"));
+        assertTrue(ASTEvaluationOnly.instanceOf(query));
+    }
+    
+    @Test
+    public void testEqDoNothing() throws ParseException {
+        ASTJexlScript query = FieldToFieldComparisonVisitor.forceEvaluationOnly(JexlASTHelper.parseJexlQuery("FOO == 'bar'"));
+        assertFalse(ASTEvaluationOnly.instanceOf(query));
+    }
+    
+    @Test
+    public void testEqDoNothingFieldsToLiteral() throws ParseException {
+        ASTJexlScript query = FieldToFieldComparisonVisitor.forceEvaluationOnly(JexlASTHelper.parseJexlQuery("(FOO || BAR).min().hashCode() == 0"));
+        assertFalse(ASTEvaluationOnly.instanceOf(query));
+    }
+    
+    @Test
+    public void testEqDoNothing3() throws ParseException {
+        ASTJexlScript query = FieldToFieldComparisonVisitor.forceEvaluationOnly(JexlASTHelper.parseJexlQuery("(UUID =~ 'C.*?' || UUID =~ 'S.*?')"));
+        System.out.println(JexlStringBuildingVisitor.buildQuery(query));
```

Comment:
Remove print statement. 

These tests look good. A general comment when it comes to testing visitors that rewrite the query tree, in the past I've delegated to a test method like `test(query, expected)` , which then runs the visitor and makes any required asserts. The bonus there is you can visually see the input and expected output right, for example...
```
query = "FOO == FOO2"
expected = "((_eval_) && (FOO == FOO2))"
test(query, expected)
```
Not requesting this as a change, just offering it as an option in the future.

Reply:
I removed the print statement. That label changed recently, so I didn't want to look for specific query. That's a good tip for writing unit tests in the future though. I will keep that in mind.